### PR TITLE
Support MetaStore without config file

### PR DESCRIPTION
**Pull request by drostron**
Fixes: #2137

### CODE REVIEW

File: interface/src/main/scala/quasar/main/metastore.scala [hunk 1]
```diff
@@ -132,29 +131,31 @@ object metastore {
...
-      mountingsConfigDecodeJson.decodeJson(jCfg).fold(
-        { case (e, _) => taskToConnectionIO(Task.fail(new RuntimeException(e.shows))) },
-        _.cata(
-          m => m.toMap.toList.traverse {
-            case (p, m) => MetaStoreAccess.insertMount(p, m)
-          }.void,
-          ().η[ConnectionIO]))
-
-    val op: ConnectionIO[Json] =
+      jCfg.cata(
```
Comment (wemrysi) [inline interface/src/main/scala/quasar/main/metastore.scala]: Replacing both `cata`s in this block with `traverse_` will get rid of both of the `().η[ConnectionIO]`.

File: interface/src/main/scala/quasar/main/metastore.scala [hunk 2]
```diff
@@ -132,29 +131,31 @@ object metastore {
...
+        ().η[ConnectionIO])
+
+    val op: ConnectionIO[Option[Json]] =
       for {
         ver <- schema.readVersion
         _   <- schema.updateToLatest
         r   <- ver.isEmpty.whenM(migrateMounts)
       } yield ver.isEmpty.fold(
-        jCfg.obj.cata(o => jObject(o - mntsFieldName), jCfg),
+        jCfg ∘ (j => j.obj.cata(o => jObject(o - mntsFieldName), j)),
```
Comment (wemrysi) [inline interface/src/main/scala/quasar/main/metastore.scala]: `jCfg ∘ (_ withObject (_ - mntsFieldName))` is a bit more straightforward.

File: interface/src/main/scala/quasar/main/package.scala [hunk 3]
```diff
@@ -367,8 +367,10 @@ package object main {
     schema: Schema[A], tx: Transactor[Task], cfgFile: Option[FsFile]
   ): MainTask[Unit] =
     for {
-      j  <- ConfigOps.jsonFromFile(cfgFile).leftMap(_.shows)
+      j  <- EitherT(ConfigOps.jsonFromFile(cfgFile).fold(
+              e => ConfigError.fileNotFound.getOption(e).cata(κ(none.right), e.shows.left),
+              _.some.right))
       jʹ <- metastore.initUpdateMetaStore(schema, tx, j)
-      _  <- EitherT.right(ConfigOps.jsonToFile(jʹ, cfgFile))
+      _  <- EitherT.right(jʹ.cata(ConfigOps.jsonToFile(_, cfgFile), Task.now(())))
```
Comment (wemrysi) [inline interface/src/main/scala/quasar/main/package.scala]: Same here, `cata` -> `traverse_`.

**Comment by drostron** [comment]
@wemrysi Excellent, I halfway thought you might have a few ⛳️ tips.

**Comment by drostron** [comment]
@wemrysi Ready for another look

**Pull request merged**